pkg/kv/kvserver/rangefeed/catchup_scan.go

```diff
@@ -94,6 +104,41 @@ func (i *CatchUpIterator) CatchUpScan(
 	withDiff bool,
 	outputFn outputEventFn,
 ) error {
+	// We emit any MVCC range tombstones first, so that the caller can use them to
+	// potentially discard point keys below them.
+	//
+	// Note that MVCC range tombstones are currently experimental, and will only
+	// be used when storage.CanUseExperimentalMVCCRangeTombstones() is enabled.
```

Comment:
Trying to build a better mental model: the caller is responsible for building some sort of "frontier" so that it can filter out anything that falls in those ranges w/ lower timestamp?

Reply:
It's up to the caller how they want to handle these. The primary use-case we have now is streaming replication, which will simply ingest these range tombstones into the target cluster as-is, without doing anything fancy.

Some time in the future we may want to give e.g. SQL access to use these range tombstones when doing ranged deletions, at which point we'll have to figure out how to emit them across CDC -- possibly by simply emitting a new event type and leave it to the consumer to handle as appropriate. In principle, these are no different than point tombstones -- some consumers may want the past history below those points, while others may want to simply discard the old values. The same applies to range tombstones, with the caveat that we won't emit diffs for them.